services/service.go

```diff
@@ -274,17 +279,32 @@ func NewService(
 // getIterationsForLimitQuery is used to get iterations of querying data from the contract with given rpc limit for blocks
 // and latest block number. Limit by default (if given limit is 0) is set to 20 000 blocks
 func (s *Service) getIterationsForLimitQuery(limit uint64) (iterations uint64, lastBlock uint64, err error) {
-	lastBlock, err = s.rpcClient.BlockNumber(context.Background())
-	if err != nil {
-		logger.Log().WithField("layer", "Service-getIterationsForLimitQuery").Errorf("get latest block rpc error: %v", err.Error())
-		return 0, 0, errors.GetRPCProviderErr(err, "BlockNumber")
+	return s.getIterationsForQuery(0, 0, limit)
+}
+
+func (s *Service) getIterationsForQuery(fromBlock uint64, toBlock uint64, limit uint64) (iterations uint64, lastBlock uint64, err error) {
+	if fromBlock == 0 {
+		fromBlock = s.perpsMarketFirstBlock
```

Comment:
this could be an issue.
The code is used by many methods. 
In some places 
s.perpsMarketFirstBlock 
is required and in some 
s.coreFirstBlock

Please.. maybe you can create two methods

getIterationsForLimitQueryPerps(limit uint64)
getIterationsForLimitQueryCore(limit uint64)
and substitute everywhere getIterationsForLimitQuery 

the logic goes like this
_Надо ориентироваться на названия функций чтобы понять что надо кор или перпс.
Либо на то какие контракты используются Внутри метода_

Reply:
the `s.perpsMarketFirstBlock` was used in `getIterationsForLimitQuery`, so I haven't changed this logic and compatibility is preserved.
Yeah, agree that if we want to extend this approach to other functions `Retrieve*` we need separate methods `getIterationsForQueryPerps` and `getIterationsForQueryCore`

Reply:
introduced new methods 
getIterationsForLimitQueryPerps(limit uint64)
getIterationsForLimitQueryCore(limit uint64)